code/statsd/src/main/scala/com/avast/datadog4s/StatsDMetricFactory.scala

```diff
@@ -5,8 +5,13 @@ import com.avast.datadog4s.api.MetricFactory
 import com.avast.datadog4s.statsd.StatsDClient
 
 object StatsDMetricFactory {
-  def make[F[_]: Sync](config: StatsDMetricFactoryConfig): Resource[F, MetricFactory[F]] =
+  def makeResource[F[_]: Sync](config: StatsDMetricFactoryConfig): Resource[F, MetricFactory[F]] =
     StatsDClient
-      .make(config.statsDServer, config.queueSize)
+      .makeResource(config.statsDServer, config.queueSize)
       .map(new statsd.StatsDMetricFactory[F](_, config.basePrefix, config.sampleRate, config.defaultTags))
+
+  def makeUnsafe[F[_]: Sync](config: StatsDMetricFactoryConfig): MetricFactory[F] = {
+    val client = StatsDClient.makeUnsafe(config.statsDServer, config.queueSize)
+    new statsd.StatsDMetricFactory[F](client, config.basePrefix, config.sampleRate, config.defaultTags)
+  }
```

Comment:
im not sure about this method ... with proposed implementation, you basically created a leak because you will not be able to close the `client` object ... in order to support that case, we would need to add `close` method to metric factory, which doesn't make much sense as there is not really anything to close. The thing that should be closed is the client. 

So i would suggest not adding such method and leaving it up to the user if he needs to create the unsafe variant. Basically you could use something like the following in your app:

```scala
def makeMetricFactory(config: StatsDMetricFactoryConfig): MetricFactory[F] = {
  val client = StatsDClient.makeUnsafe(config.statsDServer, config.queueSize)
  new statsd.StatsDMetricFactory[F](client, config.basePrefix, config.sampleRate, config.defaultTags) 
}
```

what do you think? :) It's not perfect, but i don't think there is a good way to implement this without adding `def close: Unit` everywhere

Reply:
Hm good point! My use case was for long-lived web servers, so we would only need to close it when the server shuts down. I actually think adding a `close` method to `MetricFactory` like you suggested is fine. Are you against that?

Reply:
I am tenatively against it, yes. I think it makes it more confusing to know what to actually close - do i need to close client? or factory? or both? 

But i think we could offer some method on `StatsDMetricFactory` like so:

```scala
object StatsDMetricFactory {
  ...
  def fromClient[F](statsdClient: StatsDClient, config: StatsDMetricFactoryConfig): StatsDMetricFactory[F] = ...
  ...
}
```

I think this gives the user the advantage of having fields from config correctly assigned to correct fields of the constructor without poluting the api with `close` methods, user will need to create the `StatsDClient` on his on and also close it on his on. Which will hopefully lead to code that cleans up resources correctly.

Reply:
Ok, sounds good!